bpf/custom/bpf_custom.c

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright (C) 2021 Authors of Cilium */
+
+#include <bpf/ctx/skb.h>
+#include <bpf/api.h>
+
+#include "lib/common.h"
+
+/* Include custom header file (.h) containing the implementation for
+ * custom_prog(), for example:
+ *
+ *     #include "bytecount.h"
+ */
+#include "bytecount.h"
+
+__section("custom")
```

Comment:
Encode the program type (`classifier/custom`) in the section name so that `cilium/ebpf` loader can infer the type?

Reply:
I have not done so at this time, but might be open to update it in the future. I understand it would make it easier to load the program from the Go library. On the other hand, it forces us to set a specific program type, and we loose the context flexibility TC/XDP. This probably needs some more reflection...

Reply:
Agreed. The loader does have another API where you can specify the program type while loading a BPF program, but I'm still figuring out how to get the asm instructions in a more direct way.

Reply:
We could always templatize the section name as well, something like (immediately above this line):

```
#ifndef BPF_CUSTOM_PROG_NAME
#define BPF_CUSTOM_PROG_NAME "custom"
#endif
```

plus

```suggestion
__section(BPF_CUSTOM_PROG_NAME)
```